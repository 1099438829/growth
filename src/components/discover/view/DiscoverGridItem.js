import React, { Component, PropTypes } from 'react';
import { View, Text, TouchableHighlight } from 'react-native';
import Icon from '../../../icon/GrowthFont';
import AppStyle from '../../../theme/styles';

const iconsSize = 30;
const clickItemUnderlayColor = 'rgba(250,250,250, .9)';
const LEFT = 'left';
const RIGHT = 'right';

class GridItem extends Component {
  static componentName = 'GridItem';

  static propTypes = {
    gridItemStyle: View.propTypes.style,
    iconStyle: Text.propTypes.style,
    iconName: PropTypes.string.isRequired,
    iconSize: PropTypes.number.isRequired,
    titleStyle: Text.propTypes.style,
    title: PropTypes.string.isRequired,
    onclick: PropTypes.func.isRequired,
    itemUnderlayColor: PropTypes.string.isRequired,
    position: PropTypes.string.isRequired,
  };

  static defaultProps = {
    gridItemStyle: AppStyle.discoverGridItem,
    iconStyle: AppStyle.discoverTextColor,
    iconName: '',
    iconSize: iconsSize,
    titleStyle: AppStyle.discoverGridTextItem,
    title: '',
    onclick: () => {},
    itemUnderlayColor: clickItemUnderlayColor,
    position: LEFT,

  };

  render = () => (
    <TouchableHighlight
      onPress={this.props.onclick}
      style={this.props.position === RIGHT
        ? AppStyle.discoverGridPositionRight
        : AppStyle.discoverGridPositionLeft}
      underlayColor={this.props.itemUnderlayColor}
    >
      <View style={this.props.gridItemStyle} >
        <Icon name={this.props.iconName} size={this.props.iconSize} style={this.props.iconStyle} />
        <Text style={this.props.titleStyle}>{this.props.title}</Text>
      </View>
    </TouchableHighlight>
  );
}
export default GridItem;
