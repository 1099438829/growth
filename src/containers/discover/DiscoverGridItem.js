import React, { Component, PropTypes } from 'react';
import { View, Text, TouchableHighlight } from 'react-native';
import Icon from '../../icon/GrowthFont';
import AppStyle from '../../theme/styles';

const iconsSize = 30;
const itemOnClickBackgroundColor = 'rgba(255,255,255, .5)';
const LEFT = 'left';
const RIGHT = 'right';

const defaultClick = () => {};

class GridItem extends Component {
  static componentName = 'GridItem';

  static propTypes = {
    gridItemStyle: View.propTypes.style,
    iconStyle: Text.propTypes.style,
    iconName: PropTypes.string.isRequired,
    iconSize: PropTypes.number.isRequired,
    titleStyle: Text.propTypes.style,
    title: PropTypes.string.isRequired,
    onclick: PropTypes.func.isRequired,
    onClickBackgroundColor: PropTypes.string.isRequired,
    position: PropTypes.string.isRequired,
  };

  static defaultProps = {
    gridItemStyle: AppStyle.discoverGridItem,
    iconStyle: AppStyle.discoverTextColor,
    iconName: '',
    iconSize: iconsSize,
    titleStyle: AppStyle.discoverGridTextItem,
    title: '',
    onclick: defaultClick,
    onClickBackgroundColor: itemOnClickBackgroundColor,
    position: LEFT,

  };

  render = () => (
    <TouchableHighlight
      onPress={this.props.onclick}
      style={this.props.position === RIGHT
        ? AppStyle.discoverGridPositionRight
        : AppStyle.discoverGridPositionLeft}
      underlayColor={this.props.onClickBackgroundColor}
    >
      <View style={this.props.gridItemStyle} >
        <Icon name={this.props.iconName} size={this.props.iconSize} style={this.props.iconStyle} />
        <Text style={this.props.titleStyle}>{this.props.title}</Text>
      </View>
    </TouchableHighlight>
  );

}

export default GridItem;
