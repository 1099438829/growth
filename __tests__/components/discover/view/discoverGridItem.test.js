import 'react-native';
import React, {} from 'react';
import renderer from 'react-test-renderer';
import DiscoverGridItem from '../../../../src/components/discover/view/DiscoverGridItem';

it('renders correctly', () => {
  const gridItem = renderer.create(
    <DiscoverGridItem title="学习路线" iconName="icon-social" position={'left'} onclick={() => {}} />,
  );
  const gridItemJson = gridItem.toJSON();
  expect(gridItemJson).toMatchSnapshot();
});
