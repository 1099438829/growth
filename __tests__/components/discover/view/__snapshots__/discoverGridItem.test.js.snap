// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderRadius": 5,
        "flex": 1,
        "justifyContent": "center",
      },
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 30,
          },
          Object {
            "color": "#03a9f4",
          },
          Object {
            "fontFamily": "growth",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#03a9f4",
          "marginTop": 10,
        }
      }
    >
      学习路线
    </Text>
  </View>
</View>
`;
